partition/androidboot_test.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package partition_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/partition"
+)
+
+type androidbootTestSuite struct {
+}
+
+var _ = Suite(&androidbootTestSuite{})
+
+func (g *androidbootTestSuite) SetUpTest(c *C) {
+	dirs.SetRootDir(c.MkDir())
+
+	// the file needs to exist
+	partition.MockAndroidbootFile(c, 0644)
+}
+
+func (g *androidbootTestSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("")
+}
+
+func (s *androidbootTestSuite) TestNewAndroidbootNoAndroidbootReturnsNil(c *C) {
+	dirs.SetRootDir("")
```

Comment:
Except when ran on android. I would suggest you to actually keep using the temporary random directory and populate it (or not populate it perhaps) so that it has the right data for the test. This will just run it on the real system which is not what unit tests are about.